### DISCUSSION TITLE
🚀 Feature: Implement App Attest or DeviceCheck for iOS

### DISCUSSION THREAD

**Discussion by bazrafkan** [appwrite/appwrite#11518] [Ideas]
### 🔖 Feature description

Implement App Attest or DeviceCheck on iOS SDK and backend to protect server.

Establishing your app’s integrity and ensure that requests from server receives come from legitimate instances of your app.

### 🎤 Pitch

With implement App Attest or DeviceCheck on iOS SDK and backend to protect server that uses Apple servers to certify that the key belongs to a valid instance of your app. Then you use the service to cryptographically sign server requests using the certified key. Your app uses these measures to assert its legitimacy with any server requests for sensitive or premium content. 
For more details please read this [doc](https://developer.apple.com/documentation/devicecheck/establishing_your_app_s_integrity).

![image](https://user-images.githubusercontent.com/33345799/204072792-d0fa9a20-ab7e-4dbd-9b92-3a0733da897c.png)

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
Thanks for raising this issue! 🙏 This sounds interesting! We'll keep this mind for our roadmap.

**Comment by stnguyen90** [comment]
Related issue: https://github.com/appwrite/appwrite/issues/2358

**Comment by DH-555** [comment]
That's a really good idea. 

Play integrity can be also added for Android devices: https://developer.android.com/google/play/integrity

**Comment by cddqssc** [comment]
This feature is necessary because in reality, it's very easy for people to obtain the project ID and app package name. Anyone can create new users in my project, and I can't determine which users are created by my app. This makes me feel extremely frustrated

**Comment by LUK3D** [comment]
Do we have some update on this?

**Comment by barart** [comment]
+1 on this
